src/main.cpp

```diff
@@ -190,6 +192,14 @@ auto print_configuration_template() -> void
         "port": 1247,
         "zone": "<string>",
 
+        "tls": {{
+            "client_server_policy": "<string>",
+            "ca_certificate_file": "<string>",
+            "certificate_chain_file": "<string>",
```

Comment:
alpha?   not sure it matters.

Reply:
Will reorder.

And no, order doesn't matter here.

Reply:
I reordered the options, but the felt it was a little weird since some of them are for files and others control the behavior.

Perhaps grouping them like so is better:
```json
{
    "client_server_policy": "CS_NEG_REFUSE",
    "verify_server": "cert",

    "ca_certificate_file": "/path/to/file",
    "certificate_chain_file": "/path/to/file",
    "dh_params_file": "/path/to/file"
}
```

In all honesty, the order that is defined in the PR right now is fine to me.

Perhaps `client_server_policy` should be renamed to something like `negotiation_policy`?

I think `negotiation_policy` is better, but are there other phrases that express _This client requires X for TLS_?

Maybe `secure_communication_policy`?

Thoughts on all of this?

Reply:
Ordering is fine as is - documentation is what matters.

---

`tls_policy` or `negotiation_policy`... maybe... but...

Let's not change the names - these names match the iRODS docs... unless you feel they are misleading in some way.

Reply:
No, the more I look at it, the more I feel it is fine.

If `client_server_policy` was outside the `tls` stanza, that would make the property very confusing. It being part of the `tls` stanza makes it a bit more clear in what it is trying to convey.

NFSRODS doesn't use `client_server_policy` at all. Instead, it uses `ssl_negotiation_policy`.

Changing it to `client_server_negotiation_policy` is very clear to me.

Reply:
so you're suggesting changing one attribute from

`client_server_policy` to `client_server_negotiation_policy`?

Reply:
Hmm. Well, I'm trying to remove confusion. IDK if what I just proposed makes since either.

Are those property names clear to you in regard to what we're trying to express?

To me, the property is meant to convey ... _For TLS, this app will behave as XYZ._

Reply:
Ultimately, I'm fine with the current name.

The documentation helps and it sharing a name very close to what's in the irods_environment.json file isn't a bad thing either.

Can always tweak later if we want.

Reply:
I'm going to leave things as they are.

We can discuss this on the road to 1.0.